pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -446,6 +446,9 @@ public void openCursorFailed(ManagedLedgerException exception, Object ctx) {
 
         if (!subscriptionFuture.isDone()) {
             subscriptionFuture.complete(subscription);
+        } else {
+            // failed to initialize managed-cursor: clean up created subscription
+            subscriptions.remove(subscriptionName);
```

Comment:
> Should we pass also the expected future instance?

Sorry, I didn't understand. `subscriptionFuture` future instance was failed at #line-441 due to cursor initialization and `subscriptions` map has already computed `PersistentSubscription` so, we just need to clean up `PersistentSubscription` from map and returned `subscriptionFuture` is already failed. So, where should we pass `expected future instance`?